### ISSUE TITLE
Push first release

### ISSUE THREAD

**Issue by jawshooah** [OiNutter/node-build#32]
Are you ready to push an initial release? I'm setting up a custom Homebrew [tap](https://github.com/jawshooah/homebrew-nodenv) for this and [nodenv](https://github.com/OiNutter/nodenv), and it would be helpful to provide a formula that's not head-only.

**Comment by jasonkarns** [comment]
What should releases look like for node-build? Should we try to follow ruby-build's version scheme (which is just a date string in the form of `vYYYYMMDD`) or should we stick with a proper semver-based version scheme?

Assuming the latter, we can add a `package.json` (see #50) and use the proper `version` field in there. Which would also mean we can do `npm version <major | minor | patch | etc>` to bump releases instead of using ruby-build's [release script](https://github.com/sstephenson/ruby-build/blob/master/script/release).

I think we can all agree that this would keep us more in line with what node users would expect. But it increases the drift from ruby-build proper.

Personally, my vote is to use `package.json`, semver releases, and `npm version` to do the releases. (And we can always add more scripts similar to ruby-build's scripts and execute them through `npm run-script`.)

**Comment by jawshooah** [comment]
I'm cool with the `package.json` approach. Just need to make sure we duplicate the behavior of ruby-build's release script (replacing `RUBY_BUILD_VERSION` in `bin/node-build`, publishing to GitHub and Homebrew, etc).

**Comment by ianks** [comment]
Personally not a huge fan of building in NPM as a dependency. It seems like a chicken and the egg problem to me. Maybe if we keep the NPM stuff internal... but then again it just doesnt seem neccesary.

**Comment by jawshooah** [comment]
It would only be a dependency for maintainers to push new releases. Besides, the scraper scripts already require an existing node installation.

**Comment by jasonkarns** [comment]
I'm less and less excited about having releases. On one hand, I've merged enough changes from ruby-build that the release script from ruby-build will very soon be under `scripts/`. (It hasn't been tweaked for node yet, though) However, node releases drop much faster than ruby releases ever have. Which means a *lot* of PR merging for the homebrew maintainers. Personally, I think we should push for either git installations or `HEAD` homebrew installs. This would allow end users to receive the latest from node-build much quicker without the delay of a Homebrew PR.

Alternatively, we can try to incorporate the scrapers as a first-class citizen of node-build with a CLI interface. This would allow users to get new definition files without even updating node-build.

All this being said, I don't have anything against releases. And as a matter of course, I think we ought to start tagging releases. (Once we're fully up to date with the latest from ruby-build and have their release script, this will be much easier.) But I don't think that having tagged releases will actually improve the update process for end users for the reasons I mentioned above.

**Comment by jawshooah** [comment]
I believe ruby-build has a script that automatically generates a Homebrew PR when cutting a new release (not sure, on mobile right now). Could we use something like that?

Given how many PRs they already receive on a daily basis, this would be a drop in the bucket.

**Comment by jasonkarns** [comment]
I'm still assuming we'll use the release script from ruby-build to generate the PRs for homebrew. I'm just thinking that the delay between node-build release and homebrew PR merge will be a couple days. And that's assuming we cut a release immediately each time we add a new definition(s). (At the rate of node releases, this would be weekly.)

The alternative being we recommend installing as a `--head` formula, or install via git (optionally using nodenv-update). This way, users would be able to install new node definitions immediately rather than wait for us to cut a release, and wait again for the homebrew PR to be merged.

And now to argue against my own point, I can't think of a reason *not* to do releases. Anyone who wants easy access to latest/greatest can still use --head, or git or nodenv-update.

**Comment by jawshooah** [comment]
The problem with `--head` installs is that they don't work with `brew upgrade`. The user would have to specifically run `brew reinstall --head node-build` to get any updates, and would have nothing to notify them when updates were available (short of watching this repo)

**Comment by jasonkarns** [comment]
Everyone cool with just tagging current master as `5.0.0`? This gives us some latitude to retroactively tag some old releases as major.

**Comment by jawshooah** [comment]
Fine with me. As far as I'm aware, semver doesn't care where you start as long as you're consistent.

**Comment by jasonkarns** [comment]
First release pushed and tagged.

I thought about making this initial version 5.0 so that we could retroactively tag some older versions. But Realized there was no point in tagging older releases because `node-build --version` would still report whatever was hardcoded in bin/node-build.

So, since the package.json had been set at 1.0.0 to get npm-scripts, I just bumped it to 2.0.0. https://github.com/OiNutter/node-build/releases/tag/v2.0.0

Also, to leverage npm's version bumping capabilities, I updated bin/node-build to pull the version string from package.json. (It [does so with grep/cut](https://github.com/OiNutter/node-build/blob/72863e4ab17fd3bed0ef16fbab9a5cef3353e400/bin/node-build#L661) so node-build itself still has no dependency on node. I'm sure someone could improve this with awk or something. PRs welcome.)

I have a branch in progress to tweak ruby-build's release script for our needs.
